flow360/log.py

```diff
@@ -54,15 +55,108 @@ def _get_level_int(level: LogValue) -> int:
 class LogHandler:
     """Handle log messages depending on log level"""
 
-    def __init__(self, console: Console, level: LogValue):
+    def __init__(
+        self,
+        console: Console,
+        level: LogValue,
+        fname: str = None,
+        write_to_file: bool = True,
+    ):
         self.level = _get_level_int(level)
         self.console = console
+        self.backup_count = 10
+        self.max_bytes = 10000
+        self.fname = fname
+        self.write_to_file = write_to_file
 
     def handle(self, level, level_name, message):
         """Output log messages depending on log level"""
-        if level >= self.level:
+        try:
+            if self.fname is not None and self.should_roll_over(message):
+                self.do_roll_over()
+        except OSError as error:
+            self.console.log(
+                _level_print_style.get(_level_value["ERROR"], "unknown"),
+                "Fail to Rollover" + error,
+                sep=": ",
+            )
+
+        if level >= self.level and self.write_to_file:
             self.console.log(_level_print_style.get(level_name, "unknown"), message, sep=": ")
 
+    def rotate(self, source, dest):
+        """
+        Rotate a log file by renaming the source file to the destination file.
+
+        Args:
+            source (str): Path of the source log file.
+            dest (str): Path of the destination log file.
+
+        Returns:
+            None
+        """
+        if os.path.exists(source) and not os.path.exists(dest):
+            os.rename(source, dest)
+
+    def rotation_filename(self, name, counter):
+        """
+        Generate a rotated filename based on the original filename and a counter.
+
+        Args:
+            name (str): Original filename or base filename.
+            counter (int): Counter value to be included in the rotated filename.
+
+        Returns:
+            str: Rotated filename with the format "{name}{formatted_time}.{counter}".
+
+        """
+        current_time = datetime.now()
+        formatted_time = current_time.strftime("%Y-%m-%d-%H")
+        return name + formatted_time + "." + str(counter)
+
+    def do_roll_over(self):
+        """
+        Generate a rotated filename based on the original filename and a counter.
+
+        Args:
+            name (str): Original filename or base filename.
+            counter (int): Counter value to be included in the rotated filename.
+
+        Returns:
+            str: Rotated filename with the format "{name}{formatted_time}.{counter}".
```

Comment:
I think the time in the file name is not good approach. With that we can easily overflow these 10 files leaving user with many files on their system.

<img width="1036" alt="Screenshot 2023-06-28 at 17 40 47" src="https://github.com/flexcompute/Flow360/assets/83596707/0cb4156b-e911-4280-998d-e878d418f637">